InstallDotNetCore mixes x64 and x86 versions of .NET Core in the same install root - causes dotnet.exe to crash

Attempting to update to use the new runtimes install feature. The InstallDotNetCore runtimes target is mixing x64 and x86 versions in the same installation root. This breaks .NET Core for obvious reasons - you can mix CPU architectures for native binaries.

![image](https://user-images.githubusercontent.com/2696087/57547543-8dacdd80-7313-11e9-9b9b-abbc9def7ae6.png)

Repro:
1. Clone https://github.com/natemcmaster/AspNetCore-Tooling/tree/dotnet
2. Run build.cmd

cc @chcosta

I would expect the installer to know which architecture it is installing, and not to mix bitness. You cannot mix x86 and x64 in the same installation. It causes .NET Core crash with errors like
```
##[error]dotnet.exe : Failed to load the dll from 
##[error][F:\workspace\_work\1\s\.dotnet\host\fxr\3.0.0-preview6-27708-04\hostfxr.dll], HRESULT: 0x800700C1
```

--- a/src/Microsoft.DotNet.Arcade.Sdk/src/InstallDotNetCore.cs
+++ b/src/Microsoft.DotNet.Arcade.Sdk/src/InstallDotNetCore.cs
@@ -6,6 +6,7 @@ using System.Collections.Generic;
 using System.Diagnostics;
 using System.IO;
 using System.Linq;
+using System.Runtime.InteropServices;
 using System.Text;
 using System.Text.Json;
 
@@ -94,9 +95,15 @@ namespace Microsoft.DotNet.Arcade.Sdk
                                     if(version != null)
                                     {
                                         string arguments = $"-runtime \"{runtimeItem.Key}\" -version \"{version.ToNormalizedString()}\"";
-                                        if (!string.IsNullOrWhiteSpace(item.Value))
+                                        string architecture = item.Value;
+                                        if (!string.IsNullOrWhiteSpace(architecture))
                                         {
-                                            arguments += $" -architecture {item.Value}";
+                                            arguments += $" -architecture {architecture}";
+                                        }
+                                        else if (RuntimeInformation.OSArchitecture == Architecture.X86 ||
+                                                 RuntimeInformation.OSArchitecture == Architecture.X64)
+                                        {
+                                            arguments += " -architecture x64";
                                         }
                                         Log.LogMessage(MessageImportance.Low, $"Executing: {DotNetInstallScript} {arguments}");
                                         var process = Process.Start(new ProcessStartInfo()

